napari/_qt/menus/plugins_menu.py

```diff
@@ -33,18 +31,6 @@ def __init__(self, window: 'Window'):
         self._build()
 
     def _build(self, event=None):
```

Comment:
I've kept the `_build` method as it is used here:

https://github.com/napari/napari/blob/beaa98efe5cf04ba659086e7a514b2ade05277af/napari/plugins/_npe2.py#L311-L317

but not sure what the comment means. Related: https://github.com/napari/napari/pull/4865/files#r972649032

Reply:
I see a few plugin widget tests are failing but I kept the `_add_registered_widget` fun - any ideas @DragaDoncila ?

Reply:
> but not sure what the comment means with regard to how _build should change.

So, I think this is saying that once we build the plugin menu with app_model, it should update itself when plugins are enabled/disabled, so we shouldn't need to update them here. We should then also be adding the plugin menu actions & actions created from plugin widgets in `register_manifest_actions`...?

Reply:
> I see a few plugin widget tests are failing but I kept the _add_registered_widget fun - any ideas @DragaDoncila ?

Yeah this is just a testing quirk but, now that we're not adding the `Install/Uninstall Plugins` and `Plugin Errors` buttons via this menu, there's no separator anymore so [these few lines of code](https://github.com/napari/napari/blob/2a93f1faf74ede6cbd25331e39469e6e67d2f6d2/napari/_qt/_tests/test_plugin_widgets.py#L87-L91) clear your action list (and are reproduced in all the failing tests lol). I feel like we don't even need to filter the actions, since we're just checking whether a given widget name is in there?

Reply:
> We should then also be adding the plugin menu actions & actions created from plugin widgets in register_manifest_actions...?

Thanks for explaining. To clarify I can make this change in this PR (since we are moving plugins menu to use app-model)? 

Also any thoughts on: https://github.com/napari/napari/pull/4865/files#r972649032

Reply:
> I feel like we don't even need to filter the actions, since we're just checking whether a given widget name is in there

Agreed. I will update the test.